src/string.cr

```diff
@@ -160,6 +160,16 @@ class String
     end
   end
 
+  def [](str : String)
+    if start = index(str)
+      self[start...start+str.size]
+    end
+  end
+
+  def []?(str : String)
+    !!index(str)
```

Comment:
So far the convention for `#[]?` and `#[]` is that `#[]` returns the value or raises and `#[]?` returns the value or `nil`.